shopfloor/models/stock_quant_package.py

```diff
@@ -37,3 +37,9 @@ def _compute_reserved_move_lines(self):
     # destination_planned_move_line_ids
 
     # filter out done/cancel lines
+
+    @api.constrains("name")
+    def _constrain_name_unique(self):
+        for rec in self:
+            if self.search_count([("name", "=", rec.name), ("id", "!=", rec.id)]):
+                raise exceptions.UserError(_("Package name must be unique!"))
```

Comment:
couldn't a SQL constraint be used?

Reply:
I prefer not to go for it to avoid conflicts w/ setup data and especially tests.
Also, if for any reason you want to take control of the feature, you can do it in this way.